metldata/reference/path_str.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Data models"""
+
+import re
+from typing import Optional
+
+from metldata.reference.path_elements import (
+    ReferencePathElement,
+    ReferencePathElementType,
+)
+
+NAME_PATTERN = r"[A-Za-z_]+\w*"
```

Comment:
Oh OK, I see. However, I am not sure whether we would like to allow this flexibility, since we may use the class names for other thinks such as for naming collections in MongoDB, etc. In some cases, only ascii might be allowed. However, this is not the right place to deal with that. So let's change it as you proposed.